src/shims/foreign_items.rs

```diff
@@ -594,10 +594,17 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 // FIXME: Using host floats.
                 let x = f64::from_bits(this.read_scalar(args[0])?.to_u64()?);
                 let exp = this.read_scalar(args[1])?.to_i32()?;
-                // FIXME: We should use cmath if there are any imprecisions.
-                let n = x * 2.0f64.powi(exp);
+                let n = x.ldexp(exp);
```

Comment:
Where are those methods coming from now...?

Reply:
https://github.com/rust-lang/rust/pull/63386/files#diff-f60977ab00fd9ea9ba7ac918e12a8f42R613

Reply:
Oh... that's not at all what I suggested or had in mind though.^^

I said you should use the `extern` functions. Like, access it the same way the libcore test suite does.

Reply:
I literally said "import the function via `extern` instead". No idea how you construed that into adding new APIs to libstd...

Reply:
It is a long boring story